jaxley/modules/cell.py

```diff
@@ -251,8 +331,48 @@ def init_conds(self, params: Dict) -> Dict[str, jnp.ndarray]:
         }
         return cond_params
 
+    def init_conds_jax_spsolve(self, params: Dict) -> Dict[str, jnp.ndarray]:
+        """Given length, radius, and r_a, set the coupling conductances."""
+        # `Compartment-to-compartment` conductances.
+        condition = self.comp_edges["type"].to_numpy() == 0
+        source_comp_inds = np.asarray(self.comp_edges[condition]["source"].to_list())
+        sink_comp_inds = np.asarray(self.comp_edges[condition]["sink"].to_list())
+
+        conds0 = vmap(compute_coupling_cond, in_axes=(0, 0, 0, 0, 0, 0))(
+            params["radius"][source_comp_inds],
+            params["radius"][sink_comp_inds],
+            params["axial_resistivity"][source_comp_inds],
+            params["axial_resistivity"][sink_comp_inds],
+            params["length"][source_comp_inds],
+            params["length"][sink_comp_inds],
+        )
+
+        # `branchpoint-to-compartment` conductances.
+        condition = self.comp_edges["type"].to_numpy() == 1
+        sink_comp_inds = np.asarray(self.comp_edges[condition]["sink"].to_list())
```

Comment:
is there are reason for doing `np.asarray(x.to_list())` rather than just `to_numpy()`

Reply:
could potentially be done with `.groupby("type")` as well but not sure.

Reply:
Yes, because somehow `to_numpy` returns a list of Arrays here...I still have no idea why, and I was too lazy to dive into it in detail.